### PR TITLE
Added verbose flag and logging module to nxpprog

### LINKED ISSUE #16: Move nxpprog logging to use logging module

### PR DESCRIPTION
**Pull request by evanugarte**
Logging is now done in three types corresponding to INFO, DEBUG and
ERROR. The three types are specified by an enum class, and when the user
supplies a verbose flag, all DEBUG statements are written to the
console.

Closes #16 
Resolves #16

### CODE REVIEW

File: nxpprog.py [hunk 1]
```diff
@@ -397,10 +399,19 @@
...
+    DEBUG = 1
+    ERROR = 2
 
-def log(str):
-    sys.stderr.write("%s\n" % str)
-    sys.stderr.flush()
+
+def log(str, type):
+    if type == LogType.INFO:
+        logging.info(str)
```
Comment (kammce) [inline nxpprog.py]: Any reason why we don't just use the `logging.info` calls directly vs passing a string and type?

File: nxpprog.py [hunk 2]
```diff
@@ -968,15 +980,17 @@ def prepare_flash_sectors(self, start_sector, end_sector):
     def erase_sectors(self, start_sector, end_sector, verify=False):
         self.prepare_flash_sectors(start_sector, end_sector)
 
-        log("Erasing flash sectors %d-%d" % (start_sector, end_sector))
+        log("Erasing flash sectors %d-%d" % (start_sector, end_sector),
+            LogType.DEBUG)
```
Comment (kammce) [inline nxpprog.py]: This is actually useful information for the user. This should be a normal log.

File: nxpprog.py [hunk 3]
```diff
@@ -968,15 +980,17 @@ def prepare_flash_sectors(self, start_sector, end_sector):
...
 
         if self.sector_commands_need_bank:
             self.isp_command("E %d %d 0" % (start_sector, end_sector))
         else:
             self.isp_command("E %d %d" % (start_sector, end_sector))
 
         if verify:
-            log("Blank checking sectors %d-%d" % (start_sector, end_sector))
+            log("Blank checking sectors %d-%d" % (start_sector, end_sector),
+                LogType.DEBUG)
```
Comment (kammce) [inline nxpprog.py]: This is actually useful information for the user. This should be a normal log.

File: nxpprog.py [hunk 4]
```diff
@@ -1078,7 +1092,8 @@ def prog_image(self, image, flash_addr_base=0,
             flash_addr_start = image_index + flash_addr_base
             flash_addr_end = flash_addr_start + a_ram_block - 1
 
-            log("Writing %d bytes to 0x%x" % (a_ram_block, flash_addr_start))
+            log("Writing %d bytes to 0x%x" % (a_ram_block, flash_addr_start),
+                LogType.DEBUG)
```
Comment (kammce) [inline nxpprog.py]: This is actually useful information for the user. This should be a normal log.

File: nxpprog.py [hunk 5]
```diff
@@ -1138,7 +1153,8 @@ def verify_image(self, flash_addr_base, image):
             end = end_addr if end_of_sector > end_addr else end_of_sector
             length = 4 * ((end - start) // 4)
 
-            log("Verify sector %i: Reading %d bytes from 0x%x" % (sector, length, start))
+            log("Verify sector %i: Reading %d bytes from 0x%x" % (sector, length, start),
```
Comment (kammce) [inline nxpprog.py]: This is actually useful information for the user. This should be a normal log.

File: nxpprog.py (hunk 1 above)
Comment (evanugarte) [reply nxpprog.py]: Removed log function and enum types, logging directly now

File: nxpprog.py (hunk 2 above)
Comment (evanugarte) [reply nxpprog.py]: Done.

File: nxpprog.py (hunk 3 above)
Comment (evanugarte) [reply nxpprog.py]: Done.

File: nxpprog.py (hunk 5 above)
Comment (evanugarte) [reply nxpprog.py]: Done.

**Pull request merged**